### PR TITLE
HIVE-26949: Backport HIVE-26071 to branch-3

### PR DESCRIPTION
**Pull request by junlinzeng-db**
### What changes were proposed in this pull request?

This PR backports [HIVE-26071](https://issues.apache.org/jira/browse/HIVE-26071) pr https://github.com/apache/hive/pull/3233 to branch-3. No noticeable changes involved in the backports.

### How was this patch tested?

new ut added as the original backport

### CODE REVIEW

**Comment by vihangk1** [comment]
The test failure looks unrelated. I triggered http://ci.hive.apache.org/job/hive-flaky-check/629/ to check if it is a flaky test.

**Comment by vihangk1** [comment]
I confirmed with the original PR and the backport looks good to me. +1 (pending tests)

File: standalone-metastore/src/main/java/org/apache/hadoop/hive/metastore/conf/MetastoreConf.java [hunk 1]
```diff
@@ -552,6 +552,17 @@ public enum ConfVars {
...
         "The service principal for the metastore Thrift server. \n" +
             "The special string _HOST will be replaced automatically with the correct host name."),
+    THRIFT_METASTORE_AUTHENTICATION("metastore.authentication", "hive.metastore.authentication",
+            "NONE",
+            new StringSetValidator("NONE", "JWT"),
+            "Client authentication types. Currently, only used in HTTP mode. For using kerberos use \n" +
+                    "metastore.sasl.enabled \n" +
+                    "  NONE: no authentication check\n" +
+                    "  JWT:  JSON Web Token authentication via JWT token. Only supported in Http/Https mode"),
+    THRIFT_METASTORE_AUTHENTICATION_JWT_JWKS_URL("metastore.authentication.jwt.jwks.url",
```
[the remark this answers is not in the document]
Comment (junlinzeng-db) [reply standalone-metastore/src/main/java/org/apache/hadoop/hive/metastore/conf/MetastoreConf.java:562]: done

[the remark this answers is not in the document]
Comment (vihangk1) [reply standalone-metastore/src/main/java/org/apache/hadoop/hive/metastore/conf/MetastoreConf.java:562]: thank you!

**Review by vihangk1 — APPROVED** [review]

**Comment by vihangk1** [comment]
Thanks @junlinzeng-db. Merged the PR.

**Pull request merged**